docs/source/developer-docs/contrib_guide_code.rst

```diff
@@ -55,20 +62,26 @@ be able to lend you assistance.
    reside and leave a comment in the file describing what issue you are trying
    to address.
 
-5. Open a pull request to the repository with [WIP] appended to the front
+5. Open a pull request to the repository with ``[WIP]`` appended to the front
    so that the core team is aware that you are actively pursuing the issue.
    When creating a pull request, make sure that the title clearly and concisely
    described what your code does. For example, we might use the title "Updated
    error message on ExampleException". In the body of the pull request, make sure
    that you include the phrase "Closes #issue-number-here", where the issue number is
    the issue number of the issue that you are addressing in this PR.
 
+   Feel free to open the PR as early as possible, you can get sometime early
+   feedback on your approach, which might be more efficient that polishing a
+   Pull requests that would need extensive refactor afterward.
+
 6. Run the test suite locally in order to ensure that everything is properly
    configured on your system. Refer to the repository's README for information
    on how to run the test suite. This will typically require that you run the
-   ``nosetests`` command on the commandline.
+   ``nosetests`` command on the commandline. Alternatively, you can just submit
```

Comment:
Alternatively, you may submit a pull request. Our Continuous Integration system will test your code and report test results.